var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -0,0 +1,177 @@
+
+        # CPU tests
+        if '+tests' in self.spec:
+            os.chdir(os.path.join(self.stage.source_path, 'tests'))
+            make('tests', parallel=False)
+
+        # GPU tests
+        if '+tests' in self.spec and '+cuda' in self.spec:
+            os.chdir(os.path.join(self.stage.source_path, 'gpu', 'test'))
+            make('build', parallel=False)   # this target is added by the patch
+            make('demo_ivfpq_indexing_gpu', parallel=False)
+
+        os.chdir(self.stage.source_path)
+
+    # --------------------------------------------------------------------------
+    def install(self, spec, prefix):
+
+        make('install')
+
+        if '+python' in self.spec:
+
+            # faiss's suggested installation (using makefile) puts the
+            # python bindings in python prefix
+            # but, instead, we want to keep these files in the faiss prefix
+            # TODO: replace with make install once the PR is accepted
+            # https://github.com/facebookresearch/faiss/pull/1271
+            cmd = '{} setup.py install --prefix={}'
+            os.chdir(os.path.join(self.stage.source_path, 'python'))
+            os.system(cmd.format(self.spec['python'].command, self.prefix))
+
+        def _prefix_and_install(file):
+            os.system('mv {} faiss_{}'.format(file, file))
+            install('faiss_{}'.format(file), self.prefix.bin)
+
+        # CPU tests
+        if '+tests' in self.spec:
+            os.chdir(os.path.join(self.stage.source_path, 'tests'))
+            _prefix_and_install('tests')
+
+        # GPU tests
+        if '+tests' in self.spec and '+cuda' in self.spec:
+            os.chdir(os.path.join(self.stage.source_path, 'gpu', 'test'))
+            _prefix_and_install('TestGpuIndexFlat')
+            _prefix_and_install('TestGpuIndexBinaryFlat')
+            _prefix_and_install('TestGpuIndexIVFFlat')
+            _prefix_and_install('TestGpuIndexIVFPQ')
+            _prefix_and_install('TestGpuMemoryException')
+            _prefix_and_install('TestGpuSelect')
+            _prefix_and_install('demo_ivfpq_indexing_gpu')
+
+        os.chdir(self.stage.source_path)
+
+    # --------------------------------------------------------------------------
+    @run_after('configure')
+    def _fix_makefile(self):
+
+        # for v1.5.3, the makefile contains x86-specific flags
+        # so, we need to remove them for powerpc
+        # for v1.6.0 and forward, this seems to have been fixed
+
+        # TODO: didn't check < 1.5.3 (but, do we care about the older versions)
+        # TODO: should this be removed for other architectures as well?
+        #       i.e., change the condition to target != 'x86' ?
+
+        if self.version <= Version('1.5.3') and self.spec.architecture.target == 'power9le':
+            makefile = FileFilter('makefile.inc')
+            makefile.filter( 'CPUFLAGS     = -mavx2 -mf16c',
+                            '#CPUFLAGS     = -mavx2 -mf16c')
```

Comment:
We may want to always disable this, Spack injects its own optimization flags. @alalazo thoughts?

Reply:
@adamjstewart @alalazo , please advise on this one so we can move forward.

FYI, i just rebased this on v0.15.4 and tested this again. so this should be good to go.

Reply:
I think we should always disable this. Spack automatically injects microarchitecture specific optimization flags based on the target. Having something like `-mavx2` hardcoded would make the package fail to run on older targets.